### PR TITLE
Integrate RN Nightly 6/28, bumps node from 16 to 18

### PR DESCRIPTION
**Pull request by TatianaKapos**
## Description
Integration 6/28/23: https://github.com/facebook/react-native/compare/5f8bbf2bd...15d735b35
**This PR bumps node from 16 to 18**

### Notable Changes

- [Breaking - Bump minimum Node version from 16 to 18](https://github.com/facebook/react-native/commit/43a6ea983126433d3257fafb8f01dff0e96a8c3a) 
     https://github.com/nodejs/node/issues/40702 - brought in issue using localhost, switched to 127.0.0.1
- [Native ARIA Roles: Remove JS Shim](https://github.com/facebook/react-native/commit/19699ab3b529af747223b7e470ad1e84be264b86#diff-0906918a2f25af45e6977e08e02db4813c0b705063a3cf648621d16b3b37a44f) - This will require any out of tree platforms to add their own native implementation for `role` to continue to work. https://github.com/microsoft/react-native-windows/issues/11886
- [Basic implementation of IntersectionObserver](https://github.com/facebook/react-native/commit/387bd70e49f67d310764dd23a9e6edb93cc00b73)
- [Introduce SampleLegacyModule example in RNTester ](https://github.com/facebook/react-native/commit/a1b64b7f15c475415977d2f57afe55ca6c6e4881)

 ###### Microsoft Reviewers: codeflow:open?pullrequest=https://github.com/microsoft/react-native-windows/pull/11858&drop=dogfoodAlpha

### CODE REVIEW

**Review by chiaramooney — APPROVED** [review]
Looks good to me! Could you file an issue to track the shift from accessibilityRole to role change?We'll need to update that in our native code for paper and fabric.

**Comment by TatianaKapos** [comment]
> Looks good to me! Could you file an issue to track the shift from accessibilityRole to role change?We'll need to update that in our native code for paper and fabric.

Yep! Issue created: https://github.com/microsoft/react-native-windows/issues/11886 :)

File: .ado/image/rnw-img-vs2022-node18.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
+{
```
Comment (jonthysell) [inline .ado/image/rnw-img-vs2022-node18.json:1]: If you look at the other image configs, they also contain info about the base image used, etc. Can you make sure that info is in this file too? (You can probably just copy it from the rnw-img-vs2022.json file)

**Review by jonthysell — APPROVED** [review]

File: .ado/image/rnw-img-vs2022-node18.json (hunk 1 above)
Comment (TatianaKapos) [reply .ado/image/rnw-img-vs2022-node18.json:1]: added!

**Review by chiaramooney — APPROVED** [review]

**Pull request merged**